diff --git a/node_modules/homecheck/lib/Index.d.ts b/node_modules/homecheck/lib/Index.d.ts
index d5166e1..023a627 100644
--- a/node_modules/homecheck/lib/Index.d.ts
+++ b/node_modules/homecheck/lib/Index.d.ts
@@ -1,5 +1,5 @@
 export { start, run } from './Main';
-export { BaseMetaData, BaseChecker } from './checker/BaseChecker';
+export { BaseMetaData, BaseChecker, AdviceChecker, isAdviceChecker } from './checker/BaseChecker';
 export * from './matcher/Matchers';
 export { CheckerStorage } from './utils/common/CheckerStorage';
 export { Rule } from './model/Rule';
diff --git a/node_modules/homecheck/lib/Index.js b/node_modules/homecheck/lib/Index.js
index bb7d4f1..89195f9 100644
--- a/node_modules/homecheck/lib/Index.js
+++ b/node_modules/homecheck/lib/Index.js
@@ -68,3 +68,9 @@ Object.defineProperty(exports, "Utils", { enumerable: true, get: function () { r
 var toolEntry_1 = require("./tools/toolEntry");
 Object.defineProperty(exports, "runTool", { enumerable: true, get: function () { return toolEntry_1.runTool; } });
 Object.defineProperty(exports, "Tools", { enumerable: true, get: function () { return toolEntry_1.Tools; } });
+// checkers (patched)
+var BaseChecker_1 = require("./checker/BaseChecker");
+Object.defineProperty(exports, "BaseMetaData", { enumerable: true, get: function () { return BaseChecker_1.BaseMetaData; } });
+Object.defineProperty(exports, "BaseChecker", { enumerable: true, get: function () { return BaseChecker_1.BaseChecker; } });
+Object.defineProperty(exports, "AdviceChecker", { enumerable: true, get: function () { return BaseChecker_1.AdviceChecker; } });
+Object.defineProperty(exports, "isAdviceChecker", { enumerable: true, get: function () { return BaseChecker_1.isAdviceChecker; } });
diff --git a/node_modules/homecheck/lib/checker/BaseChecker.d.ts b/node_modules/homecheck/lib/checker/BaseChecker.d.ts
index 74cdc8f..0a5e2d6 100644
--- a/node_modules/homecheck/lib/checker/BaseChecker.d.ts
+++ b/node_modules/homecheck/lib/checker/BaseChecker.d.ts
@@ -16,3 +16,8 @@ export interface BaseChecker {
     codeFix?(arkFile: ArkFile, fixKey: string): boolean;
     issues: IssueReport[];
 }
+export interface AdviceChecker extends BaseChecker {
+    beforeCheck(): void;
+    afterCheck(): void;
+}
+export declare function isAdviceChecker(checker: BaseChecker): checker is AdviceChecker;
diff --git a/node_modules/homecheck/lib/checker/BaseChecker.js b/node_modules/homecheck/lib/checker/BaseChecker.js
index 6fe0445..47c29c5 100644
--- a/node_modules/homecheck/lib/checker/BaseChecker.js
+++ b/node_modules/homecheck/lib/checker/BaseChecker.js
@@ -14,3 +14,7 @@
  * limitations under the License.
  */
 Object.defineProperty(exports, "__esModule", { value: true });
+exports.isAdviceChecker = isAdviceChecker;
+function isAdviceChecker(checker) {
+    return 'beforeCheck' in checker && 'afterCheck' in checker;
+}
diff --git a/node_modules/homecheck/lib/model/Defects.d.ts b/node_modules/homecheck/lib/model/Defects.d.ts
index 5a7b961..917709b 100644
--- a/node_modules/homecheck/lib/model/Defects.d.ts
+++ b/node_modules/homecheck/lib/model/Defects.d.ts
@@ -16,7 +16,8 @@ export declare class Defects {
     fixKey: string;
     showIgnoreIcon: boolean;
     engineName: string;
-    constructor(reportLine: number, reportColumn: number, endColumn: number, description: string, severity: number, ruleId: string, filePath: string, ruleDocPath: string, disabled: boolean, checked: boolean, fixable: boolean, showIgnoreIcon?: boolean);
+    methodName: string;
+    constructor(reportLine: number, reportColumn: number, endColumn: number, description: string, severity: number, ruleId: string, filePath: string, ruleDocPath: string, disabled: boolean, checked: boolean, fixable: boolean, methodName?: string, showIgnoreIcon?: boolean);
 }
 export declare class IssueReport {
     defect: Defects;
diff --git a/node_modules/homecheck/lib/model/Defects.js b/node_modules/homecheck/lib/model/Defects.js
index ae92866..5b1692f 100644
--- a/node_modules/homecheck/lib/model/Defects.js
+++ b/node_modules/homecheck/lib/model/Defects.js
@@ -5,7 +5,7 @@ exports.engine = {
     engineName: ''
 };
 class Defects {
-    constructor(reportLine, reportColumn, endColumn, description, severity, ruleId, filePath, ruleDocPath, disabled, checked, fixable, showIgnoreIcon = true) {
+    constructor(reportLine, reportColumn, endColumn, description, severity, ruleId, filePath, ruleDocPath, disabled, checked, fixable, methodName = "", showIgnoreIcon = true) {
         this.description = '';
         this.severity = -1; // 0:info, 1:warning, 2:error
         this.ruleId = '@perforce/<checker-name>';
@@ -22,6 +22,7 @@ class Defects {
         this.description = description;
         this.severity = severity;
         this.ruleId = ruleId;
+        this.methodName = methodName;
         this.fixKey = this.reportLine + '%' + this.reportColumn + '%' + endColumn + '%' + this.ruleId;
         this.mergeKey = filePath + '%' + this.fixKey + '%' + this.description;
         this.ruleDocPath = ruleDocPath;
